sqrl-engines/sqrl-engine-core/src/main/java/com/datasqrl/engine/server/ServerEngine.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2021, DataSQRL. All rights reserved. Use is subject to license terms.
+ */
+package com.datasqrl.engine.server;
+
+import com.datasqrl.engine.ExecutionEngine;
+import com.datasqrl.engine.stream.monitor.DataMonitor;
+import java.io.Closeable;
+
+/**
+ * The server engine is a combination of the server core (the graphql engine) and the
+ * servlet that is running it. Some servlets may not support things like Java, reflection,
+ * code generation executors, etc.
+ */
+public interface ServerEngine extends ExecutionEngine {
+
+  default DataMonitor createDataMonitor() {
```

Comment:
What is a data monitor for a Server? Can we find a different name for that since we use data monitoring currently in the context of generating data statistics in discovery. You could also rename the latter if data monitor is a better fit for what you are trying to accomplish here.

Reply:
Ah sorry, I thought this was a needed interface.